src/Services/PreHashStringFactory.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace LearnositySdk\Services;
+
+use LearnositySdk\Exceptions\ValidationException;
+use LearnositySdk\Services\PreHashStrings\PreHashStringInterface;
+use LearnositySdk\Services\PreHashStrings\LegacyPreHashString;
+
+class PreHashStringFactory
```

Comment:
I'm not sure if `PreHashString` is the best name for these classes. Maybe `PlainStringManipulator` ? Not sure if that's better though

Reply:
It's not a great name, but that's the one that has been used historically, so I went with that.

https://github.com/Learnosity/learnosity-sdk-php/blob/master/src/Request/Init.php#L297-L315

It's a very specific Service that should only do PHS generation.

Reply:
Prior to the signature upgrade in most cases the pre hash string variable was called `$signature` in many places which was so confusing. Changing it to prehash string was a revelation in terms of readability but I now realise that probably only was the case for myself and others who had struggled to understand the code when it was called `$signature`.

Has a look each implementation of of hash hmac in each of the SDK languages we support just for naming convention ideas and it seems most languages refer to the data to be hashed as message or simply data which in both cases is probably too ambiguous.